### PR TITLE
What-If Tool documentation

### PR DESCRIPTION
**Pull request by jameswex**
* Motivation for features / changes

Add markdown doc for What-If Tool in TensorBoard for use on website.

### CODE REVIEW

File: docs/r2/what-if-tool.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Model Understanding with the What-If Tool Plugin
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: done

File: docs/r2/what-if-tool.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+The What-If Tool (WIT) provides an easy-to-use interface for expanding
+understanding of black-box classification and regression ML models. With the
+plugin, you can perform inference on a large set of examples and immediately
+visualize the results in a variety of ways. Additionally, examples can be
+edited manually or programatically and re-run through the model in order to
+see the results of the changes. It contains tooling for investigating model
+performance and fairness over subsets of a dataset.
+
+The purpose of the tool is that give people a simple, intuitive, and powerful
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: fi\xed

File: docs/r2/what-if-tool.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+with absolutely no code required.
+
+The tool can be accessed through TensorBoard or directly in a Jupyter or Colab
+notebook. For more in-depth details, demos, walkthroughs, and information
+specific to using WIT in notebook mode, see the
+[What-If Tool website](https://pair-code.github.io/what-if-tool).
+
+## Requirements
+
+To use WIT in TensorBoard, only two things are necessary:
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: sure

File: docs/r2/what-if-tool.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+to load. After filling this information out and clicking "Accept", WIT will
+load the dataset and run inference with the model, displaying the results.
+
+For details on the different features of WIT and how they can aid in model
+understanding and fairness investigations, see the walkthrough on the
+[What-If Tool website](https://pair-code.github.io/what-if-tool).
+
+## Demo Model and Dataset
+
+If you want to test out WIT in TensorBoard with a pre-trained model, we have a
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: fixed

File: docs/r2/what-if-tool.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+model and dataset for you to use. The model is a binary classification model
+that uses the
+[UCI Census](https://archive.ics.uci.edu/ml/datasets/census+income) dataset
+to predict whether a person earns more than $50k a year. This dataset and
+prediction task is often used in machine learning modeling and fairness
+research.
+
+Download the data:
+`wget https://storage.googleapis.com/what-if-tool-resources/uci-census-demo/adult.tfrecord`
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: Moved all this to a single zip file to download without using wget

File: docs/r2/what-if-tool.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+Download the model:
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: done

File: docs/r2/what-if-tool.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+Note you may need to run the command with `sudo` depending on your docker
+setup.
+
+Now launch tensorboard and use the dashboard drop-down to navigate to the
+What-If Tool.
+
+On the setup screen, set the inference adddress to "localhost:8500", the
+model name to "uci_income" and the path to examples to the full path to the
+downloaded `adult.tfrecord` file, then press "Accept".
+
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: added

File: docs/r2/what-if-tool.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+The purpose of the tool is that give people a simple, intuitive, and powerful
+way to explore and investigate trained ML models through a visual interface
+with absolutely no code required.
+
+The tool can be accessed through TensorBoard or directly in a Jupyter or Colab
+notebook. For more in-depth details, demos, walkthroughs, and information
+specific to using WIT in notebook mode, see the
+[What-If Tool website](https://pair-code.github.io/what-if-tool).
+
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: added

File: docs/r2/what-if-tool.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+- The dataset to be inferred by the models must be in a TFRecord file accessible
+  by the TensorBoard web server.
+
+## Usage
+
+When opening the What-If Tool dashboard in TensorBoard, you will see a setup
+screen where you provide the host and port of the model server, the name of
+the model being served, the type of model, and the path to the TFRecords file
+to load. After filling this information out and clicking "Accept", WIT will
+load the dataset and run inference with the model, displaying the results.
```
[the remark this answers is not in the document]
Comment (jameswex) [reply docs/r2/what-if-tool.md]: done

**Comment by jameswex** [comment]
@GalOshri Made updates based on your comments. Thanks.

File: docs/r2/what-if-tool.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+For details on the different features of WIT and how they can aid in model
+understanding and fairness investigations, see the walkthrough on the
+[What-If Tool website](https://pair-code.github.io/what-if-tool).
+
+## Demo model and dataset
+
+If you want to test out WIT in TensorBoard with a pre-trained model, you can
+download and unzip a pre-trained model and dataset from
+https://storage.googleapis.com/what-if-tool-resources/uci-census-demo.zip.
```
Comment (GalOshri) [inline docs/r2/what-if-tool.md]: Does this file exist already? I am not able to download it.

File: docs/r2/what-if-tool.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+setup.
+
+Now launch tensorboard and use the dashboard drop-down to navigate to the
+What-If Tool.
+
+On the setup screen, set the inference adddress to "localhost:8500", the
+model name to "uci_income" and the path to examples to the full path to the
+downloaded `adult.tfrecord` file, then press "Accept".
+
+![Setup screen for demo](./images/what_if_tool_demo_setup.png)
```
Comment (GalOshri) [inline docs/r2/what-if-tool.md]: This image should likely be quite a bit smaller.

File: docs/r2/what-if-tool.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+Some things to try with the What-If Tool on this demo include:
+- Editing a single datapoint and seeing the resulting change in inference.
+- Exploring the relationship between individual features in the dataset and
+  the model's inference results through partial dependence plots.
+- Slicing the dataset into subsets and comparing the performance between
+  slices.
+For an in-depth look at tool's features, check out the
```
Comment (GalOshri) [inline docs/r2/what-if-tool.md]: I think an extra newline is needed to have this line appear after the bullet list.

Comment (jameswex) [reply docs/r2/what-if-tool.md]: done

File: docs/r2/what-if-tool.md (hunk 11 above)
Comment (jameswex) [reply docs/r2/what-if-tool.md]: done

File: docs/r2/what-if-tool.md (hunk 10 above)
Comment (jameswex) [reply docs/r2/what-if-tool.md]: fixed link

**Comment by jameswex** [comment]
@GalOshri addressed comments

**Pull request merged**